tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx

```diff
@@ -0,0 +1,117 @@
+// Author: Vincenzo Eduardo Padulano CERN/UPV 05/2022
+
+/*************************************************************************
+ * Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RDF_RDATASETSPEC
+#define ROOT_RDF_RDATASETSPEC
+
+#include <string>
+#include <vector>
+#include <limits>
+#include <stdexcept> // std::logic_error
+#include <algorithm> // std::transform
+
+#include <RtypesCore.h>
+#include <ROOT/InternalTreeUtils.hxx>
+
+namespace ROOT {
+
+namespace RDF {
+
+struct RDatasetSpec {
+
+   struct REntryRange {
+      ULong64_t fStartEntry{0};
+      ULong64_t fEndEntry{std::numeric_limits<ULong64_t>::max()};
+      REntryRange() {}
+      REntryRange(ULong64_t endEntry) : fEndEntry(endEntry) {}
+      REntryRange(ULong64_t startEntry, ULong64_t endEntry)
+         : fStartEntry(startEntry),
+           fEndEntry(endEntry >= startEntry
+                        ? endEntry
+                        : throw std::logic_error("RDatasetSpec: fStartEntry cannot be larger than fEndEntry."))
```

Comment:
The error could be improved imho (the user does not need to know about the names of the data members).
```
The starting entry cannot be larger than the ending entry in the creation of a dataset specification.
```